fairroot/base/sim/FairModule.cxx

```diff
@@ -210,55 +218,50 @@ void FairModule::ProcessNodes(TList* aList)
         vList = new FairVolumeList();
     }
 
-    TListIter iter(aList);
-    FairGeoNode* node = nullptr;
-    FairGeoNode* MotherNode = nullptr;
-    FairVolume* volume = nullptr;
-    FairRuntimeDb* rtdb = FairRun::Instance()->GetRuntimeDb();
-    FairGeoParSet* par = static_cast<FairGeoParSet*>(rtdb->getContainer("FairGeoParSet"));
-    TObjArray* fNodes = par->GetGeoNodes();
-    while ((node = static_cast<FairGeoNode*>(iter.Next()))) {
-
+    auto rtdb = FairRun::Instance()->GetRuntimeDb();
+    auto par = static_cast<FairGeoParSet*>(rtdb->getContainer("FairGeoParSet"));
+    auto parNodes = par->GetGeoNodes();
```

Comment:
I do like `auto` a lot.

We had a private talk about giving the `auto` a "concept" attribute to say "Okay, I don't care about the actual type, but I expect some properties from that type". In the ROOT world this would probably boil down to the base type that has all the relevant properties. I think, in this case it's [`TSeqCollection`](https://root.cern/doc/v630/classTSeqCollection.html#a38b7fc74c2740f1f02326c9c65d180d3)
```suggestion
    TSeqCollection* parNodes = par->GetGeoNodes();
```

What do you think?

Reply:
I agree, that's even better. Will change it.